### PR TITLE
Unpack subscripted function

### PR DESCRIPTION
**Pull request by mrSundvoll**

### CODE REVIEW

File: xun/functions/graph.py [hunk 1]
```diff
@@ -21,7 +21,48 @@ def source_nodes(dag):
...
+                subscript = _subscript + (idx, )
+                idx += 1
+                output += (self.unpack(shape=element, _subscript=subscript), )
+            elif isinstance(element, type(Ellipsis)):
+                subscript = _subscript + (idx, )
+                idx += 1
+                output += (CallNodeSubscript(self, subscript), )
+            else:
+                raise TypeError("Invalid content in shape tuple")
+        return output
```
Comment (mrSundvoll) [inline xun/functions/graph.py:62]: @JensGM is there a better way to share only this one function between the two successive classes?

File: xun/functions/graph.py [hunk 2]
```diff
@@ -21,7 +21,48 @@ def source_nodes(dag):
     return [n for n, in_degree in dag.in_degree() if in_degree == 0]
 
 
-class CallNode:
+class Unpackable:
```
Comment (JensGM) [inline xun/functions/graph.py:24]: Make CallNodes unpackable instead :D

**Review by JensGM — CHANGES_REQUESTED** [review]
Do magic

**Review by JensGM — APPROVED** [review]
Brilliant

**Pull request merged**